### PR TITLE
127 Docket Number Search

### PR DESCRIPTION
**Pull request by sadlerw**
* refactor persistence methods to not pass in entity and be more explicit
* add useCase specs
* add docketNumber regex to caseId as a valid identifier
* add 403 and 404 return codes to error handling
* refactor custom applicationContext out of lambdas

### CODE REVIEW

File: business/src/useCases/getCase.js [hunk 1]
```diff
@@ -5,17 +5,31 @@ const {
 const { NotFoundError, UnauthorizedError } = require('../errors/errors');
 
 exports.getCase = async ({ userId, caseId, applicationContext }) => {
-  const caseRecord = await applicationContext.persistence.get({
-    entity: { caseId, entityType: 'case' },
-    applicationContext,
-  });
+  let caseRecord;
```
Comment (dknesek) [inline business/src/useCases/getCase.js]: It seems that determining whether a string is formatted such that it is a possible case identifier is a domain problem. It should either be a responsibility of a Case itself, or there should be a new `CaseIdentifier` entity created.

File: efcms-service/src/cases/getCase.test.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ chai.use(require('chai-string'));
 describe('Get case lambda', function() {
```
Comment (dknesek) [inline efcms-service/src/cases/getCase.test.js:9]: Lambdas should be humble objects. `getCase.js` is a perfect example of one that is. It doesn't do anything except map inputs and returns. No need (or real benefit) to writing tests for these. It's just work. We should stop.

File: business/src/useCases/getCase.js (hunk 1 above)
Comment (dknesek) [inline business/src/useCases/getCase.js:9]: It seems that determining whether a string is formatted such that it is a possible case identifier is a domain problem. It should either be a responsibility of a `Case` itself, or there should be a new `CaseIdentifier` entity created.

**Review by dknesek — COMMENTED** [review]
I generally like how this looks. Just a couple of comments.

File: business/src/useCases/createCase.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    {
+      documentId: 'c6b81f4d-1e47-423a-8caf-6d2fdc3d3859',
+      documentType: 'Petition',
+    },
+  ];
+
+  beforeEach(() => {});
+
+  it('Success', async () => {
```
Comment (kkoskelin) [inline business/src/useCases/createCase.test.js:23]: wouldn't mind more descriptive statements :)

File: business/src/useCases/createCase.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    applicationContext = {
+      persistence: {
+        createCase: () =>
+          Promise.resolve({ caseId: 'c54ba5a9-b37b-479d-9201-067ec6e335bb' }),
```
Comment (kkoskelin) [inline business/src/useCases/createCase.test.js:27]: Could maybe grab this caseId and stuff it into a const?

File: business/src/useCases/createDocumentMetadata.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+const createDocumentMetadata = require('./createDocumentMetadata');
+
+describe('Create case', () => {
+  let applicationContext;
+  let documentMetaData = {
+    userId: 'taxpayer',
+    documentType: 'Petition',
+  };
+
+  beforeEach(() => {});
```
Comment (kkoskelin) [inline business/src/useCases/createDocumentMetadata.test.js:11]: superfluous?

File: business/src/useCases/getCase.js [hunk 6]
```diff
@@ -5,17 +5,31 @@ const {
...
+  const isUUID =
+    caseId &&
+    /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i.test(
```
Comment (kkoskelin) [inline business/src/useCases/getCase.js]: there be dragons
regex might be overkill if we change UUID versions?

File: business/src/useCases/getCase.test.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
+const assert = require('assert');
+const { getCase } = require('./getCase');
+
+describe('Get case', () => {
+  let applicationContext;
+
+  beforeEach(() => {});
+
+  it('Success case by case id', async () => {
```
Comment (kkoskelin) [inline business/src/useCases/getCase.test.js:9]: I'm nit-picky, I know, but -- hoping that spec strings provided to `describe` and `it` statements can be strung together to make coherent phrases, maybe more like 'Get case' 'succeeds when provided a valid case id'

**Review by kkoskelin — APPROVED** [review]

File: business/src/useCases/getCase.js (hunk 6 above)
Comment (sadlerw) [reply business/src/useCases/getCase.js]: maybe - I want to keep for now

**Pull request merged**